client/client.go

```diff
@@ -923,16 +926,28 @@ func (r *NotaryRepository) RotateKey(role string, serverManagesKey bool) error {
 	}
 
 	var (
-		pubKey    data.PublicKey
-		err       error
-		errFmtMsg string
+		pubKey     data.PublicKey
+		pubKeyList data.KeyList
+		err        error
+		errFmtMsg  string
 	)
-	switch serverManagesKey {
-	case true:
+
+	switch {
+	case serverManagesKey:
 		pubKey, err = rotateRemoteKey(r.baseURL, r.gun, role, r.roundTrip)
+		pubKeyList = make(data.KeyList, 0, 1)
+		pubKeyList = append(pubKeyList, pubKey)
 		errFmtMsg = "unable to rotate remote key: %s"
-	default:
+	case !serverManagesKey && len(keyList) > 0:
+		pubKeyList = make(data.KeyList, 0, len(keyList))
+		for _, keyID := range keyList {
+			pubKey = r.CryptoService.GetKey(keyID)
+			pubKeyList = append(pubKeyList, pubKey)
+		}
```

Comment:
`r.CryptoService.GetKey(keyID)` can return `nil` for a key, though.  Should it be checked before adding to the `pubKeyList`?  Otherwise https://github.com/docker/notary/pull/942/files#diff-1cd41a2611da49ce63630d5a51bd82c5R960 might panic.

Not sure whether the key ID not existing should warrant an error, but maybe an empty `pubKeyList` should?